util/ranger/ranger.go

```diff
@@ -28,6 +28,24 @@ import (
 	"github.com/pingcap/tidb/util/codec"
 )
 
+func validInterval(sc *stmtctx.StatementContext, low, high point) (bool, error) {
```

Comment:
Why not just rewrite the `rangePointLess`?

Reply:
It's also used in other case. And use encoded value is not easy to satisfy that need.